lib/elixir/lib/map_set.ex

```diff
@@ -32,8 +32,11 @@ defmodule MapSet do
   structures: for example, see `MapSet.new/1` or `Enum.into/2`.
   """
 
-  @opaque t :: %__MODULE__{map: map}
   @type value :: term
+
+  @opaque t(value) :: %__MODULE__{map: %{optional(value) => true}}
```

Comment:
I somehow agree, but took this from [the `Keyword` module](https://github.com/elixir-lang/elixir/blob/c4092e071f8b42f5a9ad213dd8b3632918097213/lib/elixir/lib/keyword.ex#L37). Maybe we should fix both.